src/main/java/com/mongodb/jdbc/BsonTypeInfo.java

```diff
@@ -175,9 +231,111 @@ public static int getNumPrecRadix(String typeName) throws SQLException {
             case SYMBOL_TYPE_NAME:
             case TIMESTAMP_TYPE_NAME:
             case UNDEFINED_TYPE_NAME:
+            case NULL_TYPE_NAME:
             case BSON_TYPE_NAME:
                 return 0;
         }
         throw new SQLException("unknown bson typeName: " + typeName);
     }
+
+    public static Integer getDecimalDigits(String typeName) throws SQLException {
+        switch (typeName) {
+            case DECIMAL_TYPE_NAME:
+                return 34;
+            case DOUBLE_TYPE_NAME:
+                return 15;
+            case INT_TYPE_NAME:
+            case LONG_TYPE_NAME:
+            case BINDATA_TYPE_NAME:
+            case STRING_TYPE_NAME:
+            case BOOL_TYPE_NAME:
+            case DATE_TYPE_NAME:
+            case ARRAY_TYPE_NAME:
+            case OBJECT_TYPE_NAME:
+            case OBJECTID_TYPE_NAME:
+            case DBPOINTER_TYPE_NAME:
+            case JAVASCRIPT_TYPE_NAME:
+            case JAVASCRIPTWITHSCOPE_TYPE_NAME:
+            case MAXKEY_TYPE_NAME:
+            case MINKEY_TYPE_NAME:
+            case REGEX_TYPE_NAME:
+            case SYMBOL_TYPE_NAME:
+            case TIMESTAMP_TYPE_NAME:
+            case UNDEFINED_TYPE_NAME:
+            case NULL_TYPE_NAME:
+            case BSON_TYPE_NAME:
+                return null;
+        }
+        throw new SQLException("unknown bson typeName: " + typeName);
+    }
+
+    public static Integer getCharOctetLength(String typeName) throws SQLException {
```

Comment:
These are the values we've been returning for MySQL. Not sure if they are exactly what we're supposed to be returning here...

Reply:
Both Mysql and MongoSQL are retrieving BSON data, so it should be the same, no?
If when executing a query with MongoSQL you have extra info which tell you the size of a char column, then you can use it to fine-tune this.